### ISSUE TITLE
Out of memory when using continuous curves

### ISSUE THREAD

**Issue by tnguyenh** [gama-platform/gama#2080]
### Steps to reproduce
1. display a continuous curve with a long simulation time

2.

3.

### Expected behavior
the curve should be displayed whatever the number of points.

### Actual behavior
using the attached gaml file, there is an out of memory error after about 95 cycles. Using an every(10), the memory error happens at around 300 cycles, so displaying the curve is the reason of the error. 

Depending on the integration step, there can be a huge number of points to display. However the number of points you really need to display actually depends on the chart window size: if the chart is displayed in a 200 px width window, 200 points should be enough, no matter if you are at cycle 1 or cycle 1000. 

Something like adding a facet every(cycle/200) would do the trick, but it would be better if it is transparent to the end user.

[ContinuousCurves.txt](https://github.com/gama-platform/gama/files/665472/ContinuousCurves.txt)

### System and version 
Gama 1.7, win 7

**Comment by AlexisDrogoul** [comment]
The problem described here was related to #2095 . The problem was not the plot in itself, but the historic of values kept in memory and the allocation of large StringBuilders. I have already submitted the patch (in commit 3b933747fc79ba1bfa956885fcd37172cc1b3cdc), which allows me to run your model for 300 steps without too much problems (~1.5G of memory is used). I have also released a new commit (4104eed5f8b5b36c3cd176ee6078a83c0d5366e0) that introduces an option (both a general preference and a new facet `memorize:` for `chart` to turn off the memorisation of values). Can be handy when using plots only for quickly verifying values or demonstration. 

Note that dynamic decimation (what you propose above) is not yet available. I had made it available once (18 months ago, something like this), but it was before Philippe released the new architecture, and I did not follow up. Maybe I'll make an effort ! Note also that you can use the `x_range:` facet to restrict the "window" of the plot (passing it a point with a min and a max, or a list, or an absolute number), in case you only care about the last Nth values (this will speed up considerably the drawing).

For the moment, given the 2 commits, I consider this issue as fixed. Please reopen if it is not your case.

**Comment by AlexisDrogoul** [comment]
As a side note, you can easily implement decimation by using the `every` operator on both series (x and y):

```
			chart 'chartcontinuous' type: series background: rgb('lightGray') memorize: false
			x_serie: (first(my_SEIR_maths).t[]) every 100
			size:{1.0,0.5} position:{0.0,0.0}{
				data "s_mathsc" value: (first(my_SEIR_maths).Sm[]) every 100  color: #red marker: false;
			}
```
